src/realm/object-store/sync/app.cpp

```diff
@@ -835,8 +835,12 @@ void App::log_in_with_credentials(const AppCredentials& credentials, const std::
                 return completion(nullptr,
                                   AppError(ErrorCodes::BadToken, "Could not log in user: received malformed JWT"));
             }
-            switch_user(user);
-            get_profile(user, std::move(completion));
+
+            get_profile(user, [this, completion = std::move(completion)](const std::shared_ptr<User>& user,
```

Comment:
I haven't worked with C++ in a little while, tried to get it to be equivalent but let me know if this move, etc. doesn't make sense